src/components/MessageManager/MessageManager.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+import "./MessageManager.css";
+import MessagingWindow from "../MessagingWindow/MessagingWindow";
+
+import Firebase from "firebase";
+
+class MessageManager extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      friends: [],
+      friendClicked: {},
+      user: this.props.user
+    };
+  }
+
+  componentDidMount() {
+    this.friendsRef = Firebase.database().ref(
+      "users/" + this.props.user + "/friends/"
+    ); //reference to friends
+    this.friendsRef.on("value", snapshot => {
+      this.setState(
+        {
+          friends: snapshot.val() //gives complete list of friends
+        },
+        () => {
+          let newClickedList = {};
+          Object.keys(this.state.friends).map(
+            (f, i) => (newClickedList[f] = false)
+          );
+          this.setState({
+            friendClicked: newClickedList
+          });
+        }
+      );
+    });
+  }
+
+  handleFriendClick = f => {
+    let prevFriendClicked = this.state.friendClicked;
```

Comment:
This should be written using the version of setState that takes a callback:
```
this.setState(prevState => {
  let prevFriendClicked = prevState.friendClicked;
  prevFriendClicked[friend] = !prevFriendClicked[friend];
  return { friendClicked: prevFriendClicked };
});
```
This is a something very particular to react, but you can read up on it [here](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).

The tl;dr is that it's not guaranteed that the prevFriendClicked has updated to the actual previous state, so we need to do all of that asynchronously if we're updating state based off of the previous state.

Reply:
Thank you!! That is a really important piece of information to remember!